src/hotspot/share/opto/library_call.cpp

```diff
@@ -3927,7 +3926,7 @@ bool LibraryCallKit::inline_native_Class_query(vmIntrinsics::ID id) {
     break;
   case vmIntrinsics::_getClassAccessFlags:
     prim_return_value = intcon(JVM_ACC_ABSTRACT | JVM_ACC_FINAL | JVM_ACC_PUBLIC);
-    return_type = TypeInt::INT;  // not bool!  6297094
+    return_type = TypeInt::CHAR;  // not bool!  6297094
```

Comment:
```suggestion
    return_type = TypeInt::CHAR;
```
The comment is ancient and not needed